openquake/risklib/riskinput.py

```diff
@@ -0,0 +1,382 @@
+
+    def build_input_from_ruptures(self, sitecol, assets_by_site, ses_ruptures,
+                                  gsims, trunc_level, correl_model, eps_dict):
+        """
+        :param imt: an Intensity Measure Type
+        :param hazards_by_site: an array of hazards per each site
+        :param assets_by_site: an array of assets per each site
+        :returns: a :class:`RiskInput` instance
+        """
+        imt_taxonomies = list(self.get_imt_taxonomies())
+        return RiskInputFromRuptures(
+            imt_taxonomies, sitecol, assets_by_site, ses_ruptures,
+            gsims, trunc_level, correl_model, eps_dict)
+
+    def gen_outputs(self, riskinputs, rlzs_assoc):
+        """
+        Yield the outputs generated by each getter and loss type for each
+        workflow, as a list of values, one for each realization.
+
+        :param riskinputs: a list of riskinputs with consistent IMT
+        :param rlzs_assoc: a RlzsAssoc instance
+        """
+        for riskinput in riskinputs:
+            out = self.gen_output(riskinput, rlzs_assoc)
+            # out is a dict taxonomy, loss_type -> result_by_rlz
+            for taxonomy, loss_type in sorted(out):
+                yield out[taxonomy, loss_type]
+
+    def gen_output(self, riskinput, rlzs_assoc):
+        """
+        :param riskinput: RiskInput instance
+        :param rlzs_assoc: a RlzsAssoc instance
+        :returns: a map taxonomy, loss_type -> results by realization
+        """
+        triples = zip(*riskinput.get_all())
+        output = {}
+        for imt, taxonomies in riskinput.imt_taxonomies:
+            for taxonomy in taxonomies:
+                assets, hazards, epsilons = [], [], []
+                for asset, hazard, epsilon in triples:
+                    if asset.taxonomy == taxonomy:
+                        assets.append(asset)
+                        hazards.append(hazard[imt])
+                        epsilons.append(epsilon)
+                if not assets:
+                    continue
+                # hazards is a list of dictionaries key -> array
+                hazards_by_rlz = rlzs_assoc.collect_by_rlz(hazards)
+                workflow = self[imt, taxonomy]
+                for loss_type in workflow.loss_types:
+                    # the same taxonomy contributes to two IMTs??
+                    assert (taxonomy, loss_type) not in output, (
+                        taxonomy, loss_type)
+                    out_by_rlz = {}
+                    for rlz in rlzs_assoc.realizations:
+                        haz = hazards_by_rlz[rlz]
+                        out_by_rlz[rlz] = workflow(
+                            loss_type, assets, haz, epsilons, riskinput.tags)
+                    output[taxonomy, loss_type] = out_by_rlz
+        return output
+
+
+class RiskInput(object):
+    """
+    Contains all the assets and hazard values associated to a given
+    imt and site.
+
+    :param imt: Intensity Measure Type string
```

Comment:
Please remove these params descriptions from here and add a docstring to __init__